hudi-common/src/main/java/org/apache/hudi/common/model/HoodieCommitMetadata.java

```diff
@@ -137,6 +147,16 @@ public WriteOperationType getOperationType() {
     return fullPaths;
   }
 
+  public HashMap<String, String> getFileIdAndFullPathsByPartitionPath(String basePath, String partitionPath) {
```

Comment:
In adding new APIs to shared components we should always strive for making sure that the set of APIs is bounded in size and orthogonal (ie non-overlapping b/w each other).

This method heavily overlaps with `getFileIdAndRelativePathsByPartitionPath`, let's keep just the other one